app/controllers/quizzes_controller.rb

```diff
@@ -44,4 +41,15 @@ def destroy
   def quiz_params
     params.expect(quiz: %i[title description])
   end
+
+  def find_quiz
+    @quiz = if %w[edit update destroy].include?(action_name)
+              current_user.quizzes.find(params[:id])
+            else
+              Quiz.find(params[:id])
+            end
```

Comment:
I'd imagine we'd always an authenticated user to access a quiz in the admin interface?